File: game/app/object/switch.rb
class Switch < Trigger
  def initialize(col, row, args)
    super(col, row, args, col * TILE_SIZE + 4, row * TILE_SIZE + 4, 2, 2, 'object/switch', 2, 1, img_gap: Vector.new(-3, -3))
  end

  def activate
    @img_index = 1
    super
  end
end

File: src/object/switch.lua
Switch = setmetatable({}, Trigger)
Switch.__index = Switch

function Switch.new(id, col, row, args)
  local self = Trigger.new(id, col, row, args, 7, 7, 2, 2, "object/switch" .. (args[2] or "1"), Vector.new(-5, -5), 2, 1)
  setmetatable(self, Switch)
  self.serializable_attrs = {"img_index", "active"}
  return self
end

function Switch:activate(activator)
  self.img_index = 2
  return Trigger.activate(self, activator)
end
